PostHog/posthog
# feat(settings): staff users

## Problem
As we introduced more settings that can be managed through the PostHog UI by staff users, we're presented with the problem of managing staff users.


## Changes
- Allows viewing, adding & removing staff users (_follows UI conventions on share dashboard modal & invite teammates_).
    <img width="1221" alt="" src="https://user-images.githubusercontent.com/5864173/157899623-c00d2133-e222-4f65-9c33-875336b57143.png">
    <img width="1224" alt="" src="https://user-images.githubusercontent.com/5864173/157899634-4835e046-9ccc-4dac-bfdd-09378aea58a6.png">
- Properly handles removing yourself as a staff user.
    <img width="476" alt="" src="https://user-images.githubusercontent.com/5864173/157899740-99f4ff8f-a61f-4bf4-993a-2cee43783e06.png">


## How did you test this code?
- Add & remove other staff users.
- Remove yourself and see the staff users and instance settings page disappear.
